File: C++/2952-Minimum Number of Coins to be Added.cpp
class Solution {
public:
    int minimumAddedCoins(vector<int>& coins, int target) {
        sort(coins.begin(), coins.end());
        int res = 0, reachable = 0;
        for (int coin:coins){
            while (reachable + 1 < coin && reachable < target){
                reachable += reachable+1;
                res++;
            }
            reachable += coin;
            if (reachable >= target) return res;
        }
        
        while (reachable < target){
            reachable += reachable + 1;
            res++;
        }
        return res;
    }
};

File: 2952-Minimum Number of Coins to be Added.py
class Solution:
    def minimumAddedCoins(self, coins: List[int], target: int) -> int:
        coins.sort()
        farthest = 0
        res = 0
        
        for coin in coins:
            while farthest + 1 < coin and farthest < target:
                res += 1
                farthest += farthest+1
            
            farthest += coin
            
            if farthest >= target:
                return res
            
        while farthest < target:
            farthest += farthest+1
            res += 1
        
        return res